src/main/webapp/script.js

```diff
@@ -23,24 +23,28 @@ google.setOnLoadCallback(getChart)
 
 async function getYouTubeComments() { 
   const urlInput = document.getElementById('link-input');
-  const url = cleanseUrl(urlInput.value);
+  const url = extractYouTubeUrl(urlInput.value);
   const response = await fetch("/YouTubeComments?url="+url);
   const comments = await response.json();
   return comments;
 }
 
 /*
- * Extracts video id from full url
- */ 
-function cleanseUrl(url) {
-  // Split web address from parameters, extract first parameter
-  // TODO: Add checks to make this work if video is not first parameter.
-  var videoId = url.split("?");
-  videoId = (videoId.length > 1) ? videoId[1].split("&")[0] : videoId[0];
-
-  // If param name present, remove it to isolate video Id.
-  videoId = videoId.replace("v=", "");
-  
+ * Extracts video ID from the full URL
+ * Method derived from: 
+ * https://stackoverflow.com/questions/28735459/how-to-validate-youtube-url-in-client-side-in-text-box
```

Comment:
This comment is not helpful, as people are not likely to click on a link to see why the code is written in a particular way(they will also need to spend more time understanding the question asked in the link and look through the answers). I would suggest put a comment to explain exactly what the regular expression does instead.